### PR TITLE
Patch for caliper +cuda~shared spack build

### PR DESCRIPTION
**Pull request by adayton1**
The static libraries for cupti are not installed with the CUDA toolkits on LC, so we have to use the shared libraries.

### CODE REVIEW

File: scripts/spack/packages/caliper/package.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    depends_on('cmake',  type='build')
+    depends_on('python', type='build')
+
+    # sosflow support not yet in 2.0
+    conflicts('+sosflow', '@2.0.0:2.5')
+    conflicts('+adiak', '@:2.1')
+    conflicts('+libdw', '@:2.4')
+
+    patch('for_aarch64.patch', when='target=aarch64:')
+    # SERAC EDIT START
```
Comment (adayton1) [inline scripts/spack/packages/caliper/package.py:93]: This is the change from the built in spack package.

File: scripts/spack/packages/caliper/no_static_cupti.patch [hunk 2]
```diff
@@ -0,0 +1,13 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 3bfd08e..99a0171 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -201,7 +201,7 @@ if(WITH_CUPTI)
+           ${CUDAToolkit_TARGET_DIR}/extras/CUPTI/include)
+       list(APPEND CALI_CUPTI_INCLUDE_DIRS ${_real_cupti_include_dir})
+       list(REMOVE_DUPLICATES CALI_CUPTI_INCLUDE_DIRS)
+-      if (BUILD_SHARED_LIBS)
++      if (BUILD_SHARED_LIBS OR NOT TARGET CUDA::cupti_static)
```
Comment (adayton1) [inline scripts/spack/packages/caliper/no_static_cupti.patch:10]: I only modified this conditional.

**Review by white238 — APPROVED** [review]

**Comment by white238** [comment]
Are you going to upstream this to Caliper?

**Comment by adayton1** [comment]
> Are you going to upstream this to Caliper?

I'll create a pull request and see what they say.

**Comment by adayton1** [comment]
> > Are you going to upstream this to Caliper?
> 
> I'll create a pull request and see what they say.

https://github.com/LLNL/Caliper/pull/418

**Review by jamiebramwell — APPROVED** [review]
Thanks @adayton1 !!!

**Comment by adayton1** [comment]
> > > Are you going to upstream this to Caliper?
> > 
> > 
> > I'll create a pull request and see what they say.
> 
> [LLNL/Caliper#418](https://github.com/LLNL/Caliper/pull/418)

Caliper merged in that pull request!

**Pull request merged**